### ISSUE TITLE
[Ch 09-11][Notebook][Minor] 셋업 셀에 torch 버전만 있고 transformers/datasets 버전이 없어 수치 드리프트 원인 확정 불가

### ISSUE THREAD

**Issue by modesta-jihye-lee** [yoon-gu/neuqes-101#269]
## 대상

- **챕터**: Ch 09 · Ch 10 · Ch 11 (`09_bert_regression` / `10_bert_binary_sigmoid` / `11_bert_binary_softmax`)
- **산출물**: Notebook — 셋업 셀
- **기준**: 사용자가 2026-09-12 Colab T4에서 직접 실행한 세 노트북(실행 시간 Ch09 126초 · Ch10 136초 · Ch11 165초)을 `executed/`와 셀 단위 대조

> **한 이슈로 묶은 이유**: 세 챕터가 같은 유형(셋업 셀에 `transformers`/`datasets` 버전 출력 누락)·같은 수정 경로(한 줄 추가)입니다. `cross/05`(Ch05·06 sklearn 버전 누락) 선례와 동일한 패턴입니다.

## 실제 결과

Ch09~11 세 노트북 모두 셋업 셀에서 `torch.__version__`은 출력하지만, 실제로 학습 결과(eval loss/accuracy 등)에 더 직접적인 영향을 주는 **`transformers`·`datasets` 버전은 출력하지 않습니다.**

```python
print(f"PyTorch:        {torch.__version__}")
print(f"CUDA available: {torch.cuda.is_available()}")
```

사용자가 제공한 2026-09-12 Colab 실행본과 `executed/`를 대조하면, `torch.__version__`은 동일(`2.11.0+cu128`)한데도 평가 지표가 챕터마다 갈립니다(seed=42 고정, batch/epoch 등 hyperparams 동일).

| 챕터 | 지표 | `executed/` | Colab 재실행(2026-09-12) |
|---|---|---|---|
| Ch 09 | eval_mse | 0.6539 | 0.6328 |
| Ch 09 | eval_r2 | 0.6644 | 0.6753 |
| Ch 10 | eval_accuracy (방식 A) | 0.8980 | 0.9067 |
| Ch 10 | eval_loss | 0.2759 | 0.2642 |
| Ch 11 | eval_accuracy (방식 B) | 0.9104 | 0.9129 |
| Ch 11 | eval_loss | 0.2656 | 0.2634 |

계산 관계는 보존됩니다(Ch11의 방식 A/B 비교 결론 "0.5~1%p 이내로 근접"은 두 실행 모두 성립). `torch` 버전이 같은데도 값이 갈리므로, 원인은 `transformers`(Trainer 내부 구현 변경) 또는 `datasets`(전처리·셔플 순서) 버전 차이로 추정되지만, **버전 출력이 없어 검수자가 매번 확정할 수 없습니다.**

## 기대 결과

세 챕터의 셋업 셀에 한 줄 추가:

```python
import transformers, datasets
print(f"transformers: {transformers.__version__}, datasets: {datasets.__version__}")
```

CLAUDE.md "챕터 표준 구조" §1은 "챕터가 쓰는 라이브러리로" 버전을 출력하라고 명시하며, Ch09~11은 `transformers`/`datasets`가 핵심 라이브러리인데 `torch`만 찍고 있어 조항 취지에 어긋납니다.

## 재현 절차

```bash
grep -n "__version__" 09_bert_regression/09_bert_regression.ipynb 10_bert_binary_sigmoid/10_bert_binary_sigmoid.ipynb 11_bert_binary_softmax/11_bert_binary_softmax.ipynb
# 세 파일 모두 torch.__version__ 한 줄뿐, transformers/datasets 없음
```

## 심각도

**Minor** — 수치 드리프트 자체는 계산 오류가 아니라 환경 의존(추정)이라 별도 결함이 아닙니다. 버전 출력 누락은 표준 구조 요구사항 위반이자 향후 검수자의 드리프트 원인 확정을 막습니다.

## 연결

챕터 검수 이슈 #33(Ch09) · #34(Ch10) · #35(Ch11). 선례: `cross/05`(Ch05·06 sklearn 버전 누락, 동일 패턴).
